master/buildbot/test/unit/test_fake_secrets_manager.py

```diff
@@ -0,0 +1,92 @@
+        self.master.config.secretsManagers = [FakeSecretStorage({"foo": "bar",
+                                                                 "other": "value"})]
+
+    def testGetManagerService(self):
+        secret_service_manager = SecretManager()
+        SecretManager.master = self.master
+        expectedClassName = FakeSecretStorage({"foo": "bar",
+                                               "other": "value"}).__class__.__name__
+        expectedSecretDetail = SecretDetails(expectedClassName, "foo", "bar")
+        strExpectedSecretDetail = str(secret_service_manager.get("foo"))
+        secret_result = secret_service_manager.get("foo")
+        self.assertEqual(secret_result, expectedSecretDetail)
+        self.assertEqual(secret_result.key, "foo")
+        self.assertEqual(secret_result.value, "bar")
+        self.assertEqual(secret_result.source, expectedClassName)
+        self.assertEqual(strExpectedSecretDetail,
+                         "FakeSecretStorage foo: 'bar'")
+
+    def testGetNoDataManagerService(self):
+        secret_service_manager = SecretManager()
+        SecretManager.master = self.master
+        expectedSecretDetail = SecretDetails(
+            FakeSecretStorage({"foo": "bar",
+                               "other": "value"}).__class__.__name__, "foo2",
+            None)
+        self.assertEqual(secret_service_manager.get(
+            "foo2"), expectedSecretDetail)
+
+    def testGetDataMultipleManagerService(self):
+        secret_service_manager = SecretManager()
+        self.master.config.secretsManagers = [FakeSecretStorage({"foo": "bar",
+                                                                 "other": "value"}),
+                                              OtherFakeSecretStorage({"foo2": "bar",
+                                                                      "other2": "value"},
+                                                                     props={"property": "value_prop"})
+                                              ]
+        SecretManager.master = self.master
+        expectedSecretDetail = SecretDetails(
+            OtherFakeSecretStorage({"foo2": "bar",
+                                    "other2": "value"}).__class__.__name__, "foo2",
+            "bar")
+        self.assertEqual(secret_service_manager.get(
+            "foo2"), expectedSecretDetail)
+
+    def testGetDataMultipleManagerServiceNoDatas(self):
+        secret_service_manager = SecretManager()
+        self.master.config.secretsManagers = [FakeSecretStorage({"foo": "bar",
+                                                                 "other": "value"}),
+                                              FakeSecretStorage({"foo2": "bar",
+                                                                 "other2": "value"})
+                                              ]
+        SecretManager.master = self.master
+        expectedSecretDetail = SecretDetails(
```

Comment:
I preferred to return the object because it contains the provider class name information, that will be used to prompt a log error indication from where the information has not been found.